src/parse.rs

```diff
@@ -943,22 +961,24 @@ impl<'a> Parser<'a> {
     }
 
     pub fn skip_ws(&mut self) -> Result<()> {
-        if self.src().is_empty() {
-            return Ok(());
-        }
-
-        if (self.cursor.pre_ws_cursor + self.cursor.last_ws_len) < self.cursor.cursor {
+        if (self.cursor.last_ws_len == WS_CURSOR_UNCLOSED_LINE)
+            || ((self.cursor.pre_ws_cursor + self.cursor.last_ws_len) < self.cursor.cursor)
```

Comment:
```suggestion
        if (self.cursor.last_ws_len != WS_CURSOR_UNCLOSED_LINE)
            && ((self.cursor.pre_ws_cursor + self.cursor.last_ws_len) < self.cursor.cursor)
```

Reply:
Add a test for this

* calling after the end of a token with no trailing white space should still reset the token
* calling after trailing white space should not change anything
* calling after an unclosed comment shouldn't change anything either since the document is by definition already closed and the start of the white space was at / before the start of that comment